includes/abstracts/llms-abstract-generator-posts.php

```diff
@@ -0,0 +1,787 @@
+
+		/**
+		 * Filter the status of image sideloading for the generator.
+		 *
+		 * @since [version]
+		 *
+		 * @param boolean        $enabled   Whether or not sideloading is enabled.
+		 * @param LLMS_Generator $generator Generator instance.
+		 */
+		return apply_filters( 'llms_generator_is_image_sideloading_enabled', true, $this );
+
+	}
+
+	/**
+	 * Determines if reusable block importing is enabled generator
+	 *
+	 * @since [version]
+	 *
+	 * @return boolean If `true`, importing is enabled, otherwise importing is disabled.
+	 */
+	public function is_reusable_block_importing_enabled() {
+
+		/**
+		 * Filter the status of reusable block importing for the generator.
+		 *
+		 * @since [version]
+		 *
+		 * @param boolean        $enabled   Whether or not block importing is enabled.
+		 * @param LLMS_Generator $generator Generator instance.
+		 */
+		return apply_filters( 'llms_generator_is_reusable_block_importing_enabled', true, $this );
+
+	}
+
+	/**
+	 * Load additional generator classes and other dependencies
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	protected function load_dependencies() {
+
+		// For featured image creation via `media_sideload_image()`.
+		require_once ABSPATH . 'wp-admin/includes/media.php';
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+		require_once ABSPATH . 'wp-admin/includes/image.php';
+
+	}
+
+	/**
+	 * Saves an image (from URL) to the media library and sets it as the featured image for a given post
+	 *
+	 * @since 3.3.0
+	 * @since [version] Moved from `LLMS_Generator` and made `protected` instead of `private`.
+	 *               Add a return instead of `void`; Don't import if sideloading is disabled; Use `$this->sideload_image()` sideloading.
+	 *
+	 * @param string $url_or_raw Array of raw data or URL to an image.
+	 * @param int    $post_id    WP Post ID.
+	 * @return null|false|int Returns `null` if sideloading is disabled, WP Post ID of the attachment on success, `false` on error.
+	 */
+	protected function set_featured_image( $url_or_raw, $post_id ) {
+
+		// Sideloading is disabled.
+		if ( ! $this->is_image_sideloading_enabled() ) {
+			return null;
+		}
+
+		$image_url = ( is_array( $url_or_raw ) && ! empty( $url_or_raw['featured_image'] ) ) ? $url_or_raw['featured_image'] : $url_or_raw;
+
+		if ( $image_url ) {
+
+			$id = $this->sideload_image( $post_id, $image_url, 'id' );
+			if ( ! is_wp_error( $id ) ) {
+				set_post_thumbnail( $post_id, $id );
+				return $id;
+			}
+		}
+
+		return false;
+
+	}
+
+	/**
+	 * Configure the default post status for generated posts at runtime
+	 *
+	 * @since 3.7.3
+	 *
+	 * @param string $status Any valid WP Post Status.
+	 * @return void
+	 */
+	public function set_default_post_status( $status ) {
+		$this->default_post_status = $status;
+	}
+
+	/**
+	 * Sideload an image from a url
+	 *
+	 * @since [version]
+	 *
+	 * @param int    $post_id WP_Post ID of the post where the image will be attached.
```

Comment:
```suggestion
	 * @since [version]
	 *
	 * @link https://developer.wordpress.org/reference/hooks/http_request_host_is_external/ If exporting from a local site and importing into another local site, images *will not* be side loaded as a result of this condition in the WP Core
	 *
	 * @param int    $post_id WP_Post ID of the post where the image will be attached.
```

Reply:
fine for me.